src/algebraic_geometry/structure_sheaf.lean

```diff
@@ -799,6 +799,32 @@ def basic_open_iso (f : R) : (structure_sheaf R).presheaf.obj (op (basic_open f)
   CommRing.of (localization.away f) :=
 (as_iso (show CommRing.of _ ⟶ _, from to_basic_open R f)).symm
 
+@[elementwise] lemma to_global_factors : to_open R ⊤ =
+  @category_struct.comp CommRing _ (CommRing.of R) (CommRing.of _) (CommRing.of _)
+    (show CommRing.of _ ⟶ CommRing.of _, from (algebra_map R (localization.away (1 : R))))
+  (to_basic_open R (1 : R)) ≫  (structure_sheaf R).presheaf.map (eq_to_hom (by simp)).op :=
```

Comment:
Is there any way we can use type aascriptions to avoid using `@category_struct.comp` here? This is very unreadable. Perhaps it's worth making a `CommRing.hom_of` definition, taking a `f : ring_hom A B` and producing a morphism from `CommRing.of A` to `CommRing.of B`?

Reply:
The `CommRing.hom_of` seems to work quite good.

Reply:
@erdOne I think what Adam suggested should probably be done in a separate PR, maybe for other bundled categories as well? Then we could also get rid of `@category_struct.comp` in other places. Btw, as I wrote above, even though it looks ugly with the `@category_struct.comp`, it will show up nicely with the `≫` notation when `#check`ing it or when viewing it in the online documentation.

Reply:
Should I revert the last commit then? Or to keep it and open a new PR to fix the rest of the `@category_struct.comp`s?
Though I can only find one `@category_struct.comp` outside of `src/category_theory`, i.e. the one in `Spec.lean`.

Reply:
No need to revert the last commit. Are you comfortable opening a separate PR and making this one depend on it? 
If not, then we should add a `--TODO: ...` comment above `CommRing.of_hom` explaining that analogous definitions should be made for other bundled categories as well.

Reply:
If the separated PR is just for adding `of_hom` to all the categories in `src/algebra/category/*`, I could probably do that.

Reply:
Yes, I think the separate PR should do exactly that (and replace any existing `@category.comp` similarly to what you did in this PR). Once you open such a separate PR (which will be in a separate branch), you can then merge the other branch into this one and tag this PR as dependent on the other.

If you need any help setting up the PR dependency, please don't hesitate to ask in zulip!